examples/predictors/lstm/xprize_predictor.py

```diff
@@ -91,82 +92,82 @@ def __init__(self, path_to_model_weights, data_url, cutoff_date_str):
         end_date = pd.to_datetime(end_date_str, format='%Y-%m-%d')
+        nb_days = (end_date - start_date).days + 1
 
         # Load the npis into a DataFrame, handling regions
-        npis_df = self._load_original_data(npis_csv)
-        # TODO: Current logic forecasts for each day in this npis_df. Shrink it to the requested period until
-        # we fix the logic forecast logic for the period
-        npis_df = npis_df[(npis_df.Date >= start_date) & (npis_df.Date <= end_date)]
+        npis_df = self._load_original_data(path_to_ips_file)
 
         # Prepare the output
         forecast = {"CountryName": [],
                     "RegionName": [],
                     "Date": [],
                     "PredictedDailyNewCases": []}
 
-        # For each country, each region
-        for c in self.countries:
-            cdf = self.df[self.df.CountryName == c]
-            cdf = cdf[cdf.ConfirmedCases.notnull()]
-            initial_context_input = self.country_samples[c]['X_test_context'][-1]
-            initial_action_input = self.country_samples[c]['X_test_action'][-1]
-
-            # Predictions with passed npis
-            cnpis_df = npis_df[npis_df.CountryName == c]
-            npis_sequence = np.array(cnpis_df[NPI_COLUMNS])
-
-            # Get the predictions with the passed NPIs
-            preds = self._roll_out_predictions(self.predictor,
-                                               initial_context_input,
-                                               initial_action_input,
-                                               npis_sequence)
-
-            # Gather info to convert to total cases
-            prev_confirmed_cases = np.array(cdf.ConfirmedCases)
-            prev_new_cases = np.array(cdf.NewCases)
-            initial_total_cases = prev_confirmed_cases[-1]
-            pop_size = np.array(cdf.Population)[-1]  # Population size doesn't change over time
-
-            # Compute predictor's forecast
-            pred_total_cases, pred_new_cases = self._convert_ratios_to_total_cases(
-                preds,
-                WINDOW_SIZE,
-                prev_new_cases,
-                initial_total_cases,
-                pop_size)
-            # OPTIONAL: Smooth out pred_new_cases
-            # # If window size is 7, take the previous 6 new cases so we start doing a 7 day moving average for
-            # # the first pred new cases
-            # temp_pred_new_cases = list(prev_new_cases[-(WINDOW_SIZE-1):]) + pred_new_cases
-            # smooth_pred_new_cases = self._smooth_case_list(temp_pred_new_cases, WINDOW_SIZE)
-            # # Get rid of the first window_size - 1 NaN values where
-            # # there was not enough data to compute a moving average
-            # pred_new_cases = smooth_pred_new_cases[WINDOW_SIZE-1:]
+        # For each requested geo
+        geos = npis_df.GeoID.unique()
+        for g in geos:
+            cdf = self.df[self.df.GeoID == g]
+            if len(cdf) == 0:
```

Comment:
2 cases: if don't know anything about this geo (because we don't have its population size for instance), we return 0...